storage/distributedstorage.py

```diff
@@ -64,6 +64,7 @@ def locations(self):
     validate = _location_aware(BaseStorage.validate, requires_write=True)
     get_checksum = _location_aware(BaseStorage.get_checksum)
     get_supports_resumable_downloads = _location_aware(BaseStorage.get_supports_resumable_downloads)
+    clean_partial_uploads = _location_aware(BaseStorage.clean_partial_uploads)
```

Comment:
This needs `requires_write=True` to be passed. Otherwise it will still run while in read-only mode.

Reply:
Yes good call. Added.